pvlib/temperature.py

```diff
@@ -423,3 +424,219 @@ def faiman(poa_global, temp_air, wind_speed=1.0, u0=25.0, u1=6.84):
     heat_input = poa_global
     temp_difference = heat_input / total_loss_factor
     return temp_air + temp_difference
+
+
+def _fuentes_hconv(tave, windmod, tinoct, temp_delta, xlen, tilt,
+                   check_reynold):
+    # Calculate the convective coefficient as in Fuentes 1987 -- a mixture of
+    # free, laminar, and turbulent convection.
+    densair = 0.003484 * 101325.0 / tave  # density
+    visair = 0.24237e-6 * tave**0.76 / densair  # kinematic viscosity
+    condair = 2.1695e-4 * tave**0.84  # thermal conductivity
+    reynold = windmod * xlen / visair
+    # the boundary between laminar and turbulent is modeled as an abrupt
+    # change at Re = 1.2e5:
+    if check_reynold and reynold > 1.2e5:
+        # turbulent convection
+        hforce = 0.0282 / reynold**0.2 * densair * windmod * 1007 / 0.71**0.4
+    else:
+        # laminar convection
+        hforce = 0.8600 / reynold**0.5 * densair * windmod * 1007 / 0.71**0.67
+    # free convection via Grashof number
+    # NB: Fuentes hardwires sind(tilt) as 0.5 for tilt=30
+    grashof = 9.8 / tave * temp_delta * xlen**3 / visair**2 * sind(tilt)
+    # product of Nusselt number and (k/l)
+    hfree = 0.21 * (grashof * 0.71)**0.32 * condair / xlen
+    # combine free and forced components
+    hconv = (hfree**3 + hforce**3)**(1/3)
+    return hconv
+
+
+def _hydraulic_diameter(width, height):
+    # calculate the hydraulic diameter of a rectangle
+    return 2 * (width * height) / (width + height)
+
+
+def fuentes(poa_global, temp_air, wind_speed, noct_installed, module_height=5,
+            wind_height=9.144, emissivity=0.84, absorption=0.83,
+            surface_tilt=30, module_width=0.31579, module_length=1.2):
+    """
+    Calculate cell or module temperature using the Fuentes model.
+
+    The Fuentes model is a first-principles heat transfer energy balance
+    model [1]_ that is used in PVWatts for cell temperature modeling [2]_.
+
+    Parameters
+    ----------
+    poa_global : pandas Series
+        Total incident irradiance [W/m^2]
+
+    temp_air : pandas Series
+        Ambient dry bulb temperature [C]
+
+    wind_speed : pandas Series
+        Wind speed [m/s]
```

Comment:
seems like this function would work just as well with any kind of array input, right? If so, we should make the type requirements here and in the return line `array_like` and change the last line of the function to only return a series if `poa_global` is a series.

Reply:
Unfortunately the timedelta between measurements is needed for the thermal capacitance/time lag part of the model, so I think it has to either be Series inputs or another parameter would be added for the measurement timestamps.

Reply:
ah, right. let's leave it alone then. new merge conflict (sorry)